private/bufpkg/bufcheck/bufbreaking/internal/bufbreakingcheck/bufbreakingcheck.go

```diff
@@ -313,6 +383,33 @@ func checkFieldSameOneof(add addFunc, corpus *corpus, previousField bufprotosour
 	return nil
 }
 
+// CheckFieldSamePresence is a check function.
+var CheckFieldSamePresence = newFieldDescriptorPairCheckFunc(checkFieldSamePresence)
+
+func checkFieldSamePresence(
+	add addFunc,
+	_ *corpus,
+	_ bufprotosource.Field,
+	previousDescriptor protoreflect.FieldDescriptor,
+	field bufprotosource.Field,
+	descriptor protoreflect.FieldDescriptor,
+) error {
+	previousPresence := getFieldPresence(previousDescriptor)
+	currentPresence := getFieldPresence(descriptor)
+	if previousPresence != currentPresence {
+		// otherwise prints as hex
+		numberString := strconv.FormatInt(int64(field.Number()), 10)
+		// TODO: specific label location
+		add(field, nil, field.Location(),
+			`Field %q on message %q changed presence from %q to %q.`,
+			numberString, field.ParentMessage().Name(),
```

Comment:
Same question about nil (will be whole way down for such call types)

Reply:
nil for... the `ParentMessage()`? I guess I didn't think about it since this was basically copied (nearly every field check in here is formulated exactly this way).

I guess they _all_ need to be revised to behave correctly (and avoid possible nil-dereference panics) in the face of file-level extensions.

Reply:
Refactored this to use a helper and avoid potentially de-referencing `ParentMessage()` for top-level extensions.